# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/alhvirk-web/FlyRank-Ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring. I'm choosing this because the starter dataset directly supports it — it has trend data (trend_direction, trend_pct) I can use to define an observed outcome, and the shipped pipeline already proves a model beats a hand-rule here (Precision@50: 0.24 → 0.74), so there's real signal to find.

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: which pages an editor should review or refresh first. Actor: a content editor with limited weekly review capacity. Cost of a wrong call: reviewing a page that wasn't actually declining wastes editor time; missing a page that was actually declining means a real traffic loss goes unaddressed.

Unit of analysis: one row is one content page, measured over its trailing 90-day window. Output: a ranked queue of pages ordered by priority score, not just a single yes/no flag. Why ML and not a fixed rule: this pipeline's own hand-written baseline rule only reaches Precision@50 ≈ 0.24, which is low enough to show that decline isn't driven by one clean signal — it's several things (word count, freshness, position, content type) interacting together, and that's exactly the kind of pattern ML is better suited to weigh than a single if-statement.

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import os, subprocess
if not os.path.isdir("FlyRank-Ml"):
    subprocess.run(["git", "clone", "--depth", "1", "https://github.com/alhvirk-web/FlyRank-Ml"])
os.chdir("FlyRank-Ml")

import pandas as pd
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
print(df.shape)
print(df["trend_direction"].value_counts())
print((df["trend_direction"] == "down").mean())


(30000, 44)
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64
0.5420666666666667


Out of 30,000 pages across 32 clients, 54.2% (16,262) are currently declining. That's the largest single bucket by far — this isn't an edge case, it's the majority state of content here, which makes a prioritization tool genuinely useful.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I can say: this is observed, decision-support work — it estimates a priority ranking for review, and any lift over baseline is measured on this dataset. I cannot say this predicts Google's algorithm, proves causation for why a page declined, or guarantees outcomes for pages outside this snapshot.

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.